core/src/main/scala/org/apache/spark/scheduler/Pool.scala

```diff
@@ -97,8 +99,14 @@ private[spark] class Pool(
 
   override def getSortedTaskSetQueue: ArrayBuffer[TaskSetManager] = {
     var sortedTaskSetQueue = new ArrayBuffer[TaskSetManager]
+    def targetSchedulables = { sq:ConcurrentLinkedQueue[Schedulable] =>
+      if (maxRunning != 0)
+        sq.take(maxRunning).toSeq
+      else
+        sq.toSeq
+    }
```

Comment:
this is only used once so it can be a val:
```
val targetSchedulables =
  if (maxRunning != 0) {
    schedulableQueue.take(maxRunning).toSeq
  } else {
    schedulableQueue.toSeq
  }
val sortedSchedulableQueue = targetSchedulables.sortWith(...)
```